include/glow/Backend/BackendUtils.h

```diff
@@ -120,5 +120,27 @@ class RuntimeBundle {
 };
 } // namespace runtime
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH, const Function &F);
+
+/// If \p PH is an input placeholder, \returns true.
+bool isInput(const Placeholder *PH, const Function &F);
+
+/// If \p PH is an output placeholder in the function \p F, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH, const IRFunction &F);
+
+/// If \p PH is an input placeholder in the function \p F, \returns true.
+bool isInput(const Placeholder *PH, const IRFunction &F);
+
+/// Convert placeholders to be ordered as input|inputOutput|output|neither.
+/// Packed into {Placeholder *, isInput, isOutput}.
+template <typename FUN, typename ARR>
+std::vector<std::tuple<const Placeholder *, bool, bool>>
```

Comment:
Nit: I wonder of using a small struct with 3 fields as a helper would be a cleaner and more understandable solution?

Reply:
> Nit: I wonder of using a small struct with 3 fields as a helper would be a cleaner and more understandable solution?

OK. I grouped them into a struct with 3 fields.